rskj-core/src/main/java/org/ethereum/rpc/JsonRpcWeb3FilterHandler.java

```diff
@@ -0,0 +1,71 @@
+package org.ethereum.rpc;
+
+import co.rsk.rpc.JsonRpcFilterServer;
+import co.rsk.rpc.ModuleDescription;
+import co.rsk.rpc.OriginValidator;
+import com.googlecode.jsonrpc4j.AnnotationsErrorResolver;
+import com.googlecode.jsonrpc4j.DefaultErrorResolver;
+import com.googlecode.jsonrpc4j.MultipleErrorResolver;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.*;
+import org.ethereum.rpc.exception.RskErrorResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URISyntaxException;
+import java.util.List;
+
+/**
+ * Created by ajlopez on 18/10/2017.
+ */
+
+@ChannelHandler.Sharable
+public class JsonRpcWeb3FilterHandler extends SimpleChannelInboundHandler<FullHttpRequest> {
+    private static final Logger LOGGER = LoggerFactory.getLogger("jsonrpc");
+
+    private OriginValidator originValidator;
+
+    public JsonRpcWeb3FilterHandler(String corsOrigins) {
+        try {
+            this.originValidator = new OriginValidator(corsOrigins);

```

Comment:
If this throws an exception `originValidator` will be null and we'll face `NPE`s later when `channelRead0` is called

Reply:
It would be better to receive the `OriginsValidator` instance in the constructor IMHO